test/umzug.test.ts

```diff
@@ -114,6 +114,34 @@ describe('custom context', () => {
 		expect(pending[0]).toContain('test.x.js');
 	});
 
+	test(`create with custom template async method`, async () => {
+		const syncer = fsSyncer(path.join(__dirname, 'generated/create-custom-template-async'), {});
+		syncer.sync();
+
+		const methodAsync:()=>Promise<string> = ()=> {
+			return new Promise((resolve)=>{
+				return resolve(`/* custom template async */`)
+			})
+		}
```

Comment:
You'll get lint errors here, fix with `npm run lint -- --fix`

Also this type annotation is awkward, let's drop it:

```suggestion
		const methodAsync = async () => {
			return `/* custom template async */`
		}
```

And last minor thing, let's call it `template` rather than `methodAsync`, it would be good to keep the tests as realistic as possible.